packages/snippet-bot/src/snippet-bot.ts

```diff
@@ -346,6 +379,100 @@ ${bodyDetail}`
+      const diff = await response.text();
+      const diffResult = parseDiff(diff);
+      const changes: Change[] = [];
+      let added = 0;
+      let deleted = 0;
+      for (const file of diffResult) {
+        for (const chunk of file.chunks) {
+          for (const change of chunk.changes) {
+            if (change.type === 'normal') {
+              continue;
+            }
+            // We only track add/deletion of start tags.
+            const startMatch = change.content.match(START_TAG_REGEX);
+            // TODO: change owner and repo for deletion
+            if (startMatch) {
+              if (change.type === 'add') {
+                added += 1;
+              }
+              if (change.type === 'del') {
+                deleted += 1;
+              }
+              changes.push({
+                type: change.type === 'del' ? 'del' : 'add',
+                region_tag: startMatch[1],
+                owner: change.type === 'del' ? owner : headOnwer,
+                repo: change.type === 'del' ? repo : headRepo,
+                file: change.type === 'del' ? file.from : file.to,
+                sha: change.type === 'del' ? sha : headSha,
+                line: change.ln,
+              });
+            }
+          }
+        }
+      }
+      if (changes.length === 0) {
+        return;
+      }
+
+      // Add or update a comment on the PR.
+      const prNumber = context.payload.pull_request.number;
+      let commentBody = 'Here is the summary of changes.\n';
+      if (added > 0) {
+        const plural = added === 1 ? '' : 's';
+        const summary = `You added ${added} region tag${plural}.`;
+        let detail = '';
+        for (const change of changes) {
+          if (change.type === 'add') {
+            detail += `- ${formatChangedFile(change)}\n`;
+          }
+        }
+        commentBody += formatExpandable(summary, detail);
+      }
+      if (deleted > 0) {
+        const plural = deleted === 1 ? '' : 's';
+        const summary = `You deleted ${deleted} region tag${plural}.\n`;
+        let detail = '';
+        for (const change of changes) {
+          if (change.type === 'del') {
+            detail += `- ${formatChangedFile(change)}\n`;
+          }
+        }
+        commentBody += formatExpandable(summary, detail);
+      }
+
+      const listCommentsResponse = await context.github.issues.listComments({
+        owner: owner,
+        repo: repo,
+        per_page: 50,
+        issue_number: prNumber,
+      });
+      let found = false;
+      for (const comment of listCommentsResponse.data) {
+        if (comment.body.includes(commentMark)) {
+          // We found the existing comment, so updating it
+          await context.github.issues.updateComment({
+            owner: owner,
+            repo: repo,
+            comment_id: comment.id,
+            body: `${commentMark}\n${commentBody}`,
+          });
+          found = true;
```

Comment:
I like this pattern to avoid spamming pull requests with comments.

Reply:
should we consider pulling some common patterns out into helpers in `gcf-utils` perhaps?

Reply:
ack